[TensorExpr] Implement 'call_raw' in IREval.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #57901 [TensorExpr] Implement 'call_raw' in CUDA codegen.
* **#57882 [TensorExpr] Implement 'call_raw' in IREval.**
* #57881 [TensorExpr] Minor cleanup in IREval.

Differential Revision: [D28306752](https://our.internmc.facebook.com/intern/diff/D28306752)

Commit:
[TensorExpr] Implement 'call_raw' in IREval.

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28306752).

navahgar commented on torch/csrc/jit/tensorexpr/eval.cpp:
```diff
@@ -985,22 +993,20 @@ void SimpleIREvaluator::call(const std::vector<CallArg>& args) {
+#define TYPE_CASE(Type, Name)                 \
+  case ScalarType::Name: {                    \
+    Type typed_data;                          \
+    memcpy(&typed_data, &data, sizeof(Type)); \
```
We seem to be adding an extra `memcpy` here. Isn't this increasing the overhead? Should we retain the old code for when it is called using `CallArg`?

navahgar reviewed (approved):
LGTM

Commit:
Update on "[TensorExpr] Implement 'call_raw' in IREval."


Differential Revision: [D28306752](https://our.internmc.facebook.com/intern/diff/D28306752)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28306752).

Commit:
Update on "[TensorExpr] Implement 'call_raw' in IREval."


Differential Revision: [D28306752](https://our.internmc.facebook.com/intern/diff/D28306752)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28306752).

Commit:
Update on "[TensorExpr] Implement 'call_raw' in IREval."


Differential Revision: [D28306752](https://our.internmc.facebook.com/intern/diff/D28306752)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28306752).
